### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Yerimah**
In this branch I worked on the following:
- Created classes for persons, students, and teachers with instance variables
- Fixed linter errors

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id
```
Comment (TedLivist) [inline person.rb]: - [x] For better code organization I would recommend you move these getters and setters to the top of the class.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age?
+  end
```
Comment (TedLivist) [inline person.rb]: - [x] The method `can_use_services?` should return true if the `of_age?` **OR** the `@parent_permission` variable is true, Right now it only checks for `of_age?`. Kindly refactor accordingly.

  **SUGGESTION:** You can refactor the function to be like so:
  ```
  def can_use_services?
    of_age? || @parent_permission
  end
  ```

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  private :is_of_age?
```
Comment (TedLivist) [inline person.rb]: - [x] The `:is_of_age?` method does not exist in this class. Kindly fix that. I think you meant `:of_age?`.

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'unknown')
+    @specialization = specialization
+    super(age, name)
+  end
```
Comment (TedLivist) [inline teacher.rb]: - [x] You are missing the `parent_permission` parameter. Kindly add it and also do the same in the `super` method.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Yerimah,

Good job so far!
✔️ Classes are in different files
✔️ Implementation of Student class
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
### Look here too 🛑 🛑
- [x] **Good job with the README, however, you are still missing some sections in the template. Kindly add the missing sections according to the [Microverse template](https://github.com/microverseinc/readme-template). All sections are required but you can skip the Deployment, Tests and Demo link parts if you don't have them yet.**

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Yerimah, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Yerimah** [comment]
Thank you

**Pull request merged**